demo/ARMapNavigationController.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  ARMapNavigationController.swift
+//  cv-assist-ios
+//
+//  Created by Alexander Pristavko on 3/20/18.
+//  Copyright © 2018 Mapbox. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import Mapbox
+import MapboxCoreNavigation
+import MapboxNavigation
+import MapboxDirections
+
+private let routeEdgeInsets = UIEdgeInsets(top: 100, left: 200, bottom: 100, right: 200)
+private let inset: CGFloat = 18.0
+
+class ARMapNavigationController: UIViewController {
+    
+    var completion: ((Route) -> Void)?
+    
+    private var mapView: NavigationMapView {
+        return view as! NavigationMapView
+    }
+    
+    private var selectedRoute: Route?
+    
+    override func loadView() {
+        let view = NavigationMapView(frame: .zero, styleURL: MGLStyle.darkStyleURL)
+        view.delegate = self
+        view.showsUserLocation = true
+        view.translatesAutoresizingMaskIntoConstraints = false
+        self.view = view
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.addGestureRecognizer(UILongPressGestureRecognizer(target: self, action: #selector(selectPlace)))
+        
+        view.addSubview(goButton)
+        NSLayoutConstraint.activate([
+            goButton.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
```

Comment:
That may lead to lack of inset on setups with 0 trailing safeArea inset. Add constant inset from view's `layoutGuide`.